### PR TITLE
fix(runner): fix bugs in runner batching

### PR DESCRIPTION
**Pull request by sauyon**
Now properly deals with edge cases, I think... I want to factor this out into its own module so we can reuse it, I think that would make sense. (Also we could write unit tests for the logic...)

### CODE REVIEW

File: bentoml/_internal/runner/runnable.py [hunk 1]
```diff
@@ -135,7 +135,9 @@ def get_batch_dim(idx: int) -> int:
 
         self.args = []
         self.kwargs = {}
-        for i, (name, param) in enumerate(params.items()):
+        # drop self parameter from params
+        param_items = list(params.items())[1:]
```
Comment (bojiang) [inline bentoml/_internal/runner/runnable.py:139]: not all the cases have self as the first arg

File: bentoml/_internal/runner/runner_handle/local.py [hunk 2]
```diff
@@ -16,15 +17,17 @@ class LocalRunnerRef(RunnerHandle):
...
-    def run_method(self, method_name: str, *args: t.Any, **kwargs: t.Any) -> t.Any:
-        return getattr(self._runnable, method_name)(*args, **kwargs)
+    def run_method(
+        self, __bentoml_method_name: str, *args: t.Any, **kwargs: t.Any
+    ) -> t.Any:
+        return getattr(self._runnable, __bentoml_method_name)(*args, **kwargs)
 
     async def async_run_method(
-        self, method_name: str, *args: t.Any, **kwargs: t.Any
+        self, __bentoml_method_name: str, *args: t.Any, **kwargs: t.Any
```
Comment (bojiang) [inline bentoml/_internal/runner/runner_handle/local.py:26]: 👍

File: bentoml/_internal/runner/runnable.py (hunk 1 above)
Comment (sauyon) [reply bentoml/_internal/runner/runnable.py:139]: I think it's ok to not support `staticmethod` here, though we may want to have better error messages for those cases.
